lib/plugins/aws/deployFunction/index.js

```diff
@@ -33,8 +33,10 @@ class AwsDeployFunction {
       'deploy:function:deploy': () => BbPromise.bind(this)
         .then(() => {
           if (!this.options['update-config']) {
-            this.deployFunction();
+            return this.deployFunction();
```

Comment:
Without `return` `deployFunction` and  `updateFunctionConfiguration` invoke in the exact same moment and cause issues.